internal/orchestrator/status.go

```diff
@@ -129,3 +133,123 @@ func (or *orchestrator) GetStatus(ctx context.Context) (status *core.NamespaceSt
+	filter := fb.Sort("created").Limit(1).And(
+		fb.Eq("type", core.MessageTypeDefinition),
+		fb.Eq("tag", core.SystemTagIdentityClaim),
+		fb.Eq("author", core.FireFlyOrgDIDPrefix+or.config.Multiparty.Org.Name),
+		fb.In("state", []driver.Value{core.MessageStateStaged, core.MessageStateReady, core.MessageStateSent, core.MessageStatePending}),
+	)
+	msgs, _, err := or.GetMessagesWithData(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	if len(msgs) > 0 {
+		return msgs[0], nil
+	}
+	return nil, nil
+}
+
+// Ensure the given registration message correlates with the expected type
+func (or *orchestrator) checkRegistrationType(ctx context.Context, msg *core.MessageInOut, expectedType core.IdentityType) (match bool, err error) {
+	if msg != nil && msg.InlineData != nil && len(msg.InlineData) > 0 {
+		var identity core.IdentityClaim
+		err := json.Unmarshal([]byte(*msg.InlineData[0].Value), &identity)
+		if err != nil {
+			return false, i18n.WrapError(ctx, err, coremsgs.MsgUnableToParseRegistrationData, err)
+		}
+		var did string
+		switch expectedType {
+		case core.IdentityTypeOrg:
+			did = core.FireFlyOrgDIDPrefix + or.config.Multiparty.Org.Name
+		case core.IdentityTypeNode:
+			did = core.FireFlyNodeDIDPrefix + or.config.Multiparty.Node.Name
+		default:
+			return false, i18n.NewError(ctx, coremsgs.MsgUnexpectedRegistrationType, expectedType)
+		}
+		return identity.Identity.DID == did && identity.Identity.Type == expectedType, nil
+	}
+	return false, i18n.NewError(ctx, coremsgs.MsgNoMessageInlineData, or.config.Multiparty.Org.Name)
+}
+
+func (or *orchestrator) GetMultipartyStatus(ctx context.Context) (mpStatus *core.NamespaceMultipartyStatus, err error) {
+
+	mpStatus = &core.NamespaceMultipartyStatus{
+		Enabled: or.config.Multiparty.Enabled,
+	}
+	if !or.config.Multiparty.Enabled {
+		return mpStatus, nil
+	}
+
+	status, err := or.GetStatus(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	mpStatus.Org = core.NamespaceMultipartyStatusOrg{}
+	mpStatus.Node = core.NamespaceMultipartyStatusNode{}
+	mpStatus.Contracts = &core.MultipartyContractsWithActiveStatus{
+		Active: &core.MultipartyContractWithStatus{
+			MultipartyContract: *or.namespace.Contracts.Active,
+			Status:             core.ContractListenerStatusUnknown,
+		},
+		Terminated: or.namespace.Contracts.Terminated,
+	}
+
+	log.L(ctx).Debugf("Looking up listener status with subscription ID: %s", mpStatus.Contracts.Active.Info.Subscription)
+	ok, _, listenerStatus, err := or.blockchain().GetContractListenerStatus(ctx, or.namespace.Name, mpStatus.Contracts.Active.Info.Subscription, false)
+	if !ok || err != nil {
+		return nil, err
+	}
+	mpStatus.Contracts.Active.Status = listenerStatus
+
+	if status.Org.Registered {
+		mpStatus.Org.Status = core.NamespaceRegistrationStatusRegistered
+		if status.Node.Registered {
+			mpStatus.Node.Status = core.NamespaceRegistrationStatusRegistered
+		} else {
+			msg, err := or.getRegistrationMessage(ctx)
```

Comment:
It's interesting to me that we only query "bad/stuck" registration messages. So we do _not_ populate `RegistrationMessageID` in the case that registration was successful.

I think we should either a) populate it in all cases, or b) rename the field so it's clear that it only returns the ID of a non-confirmed registration.

Reply:
Fair point - I didn't populate it in the success case purely because my use case didn't require it, but I think option a) makes for the most simple, understandable API. I'll update the PR.